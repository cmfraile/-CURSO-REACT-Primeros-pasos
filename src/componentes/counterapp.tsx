interface counterif {contador:number};
export const counter = ({contador}:counterif) => {

    return(
        <>
            <h1>Counterapp</h1>
            <h2>{contador}</h2>
        </>
    )
    
}