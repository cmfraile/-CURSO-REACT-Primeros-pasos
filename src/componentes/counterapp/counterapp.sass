.capp
    h1,h2
        text-align: center
        font-weight: 700
        font-size: 2.5rem
    button
        width: 50px
        height: 50px
        border: none
        background-color: grey
        border-radius: 20%
        font-size: 2rem
        font-weight: 700