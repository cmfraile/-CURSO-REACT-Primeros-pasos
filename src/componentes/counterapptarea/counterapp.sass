.capp
    h1,h2
        text-align: center
        font-weight: 700
        font-size: 2.5rem
    button
        min-width: 50px
        height: 50px
        border: none
        background-color: grey
        border-radius: 20%
        font-size: 1.5rem
        font-weight: 700
        margin: 10px
        padding: 10px