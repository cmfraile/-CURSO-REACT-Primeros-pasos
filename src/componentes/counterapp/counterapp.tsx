import { MouseEvent , useState } from 'react';
import './counterapp.sass';

interface counterif {contador:number};
export const counter = ({contador}:counterif) => {

    const [ contadorhook , setCounter ] = useState(contador)
    const handlevent = (event:MouseEvent) => {
        setCounter(contadorhook+1);
        //setCounter( (c) => c + 1 );
    };

    return(
        <div className='capp' >
            <h1>Counterapp</h1>
            <h2>{contadorhook}</h2>
            <button onClick={ handlevent }>+1</button>
        </div>
    )

    /*
    return(
        <>
            <h1>Counterapp</h1>
            <h2>{contador}</h2>
            <button onClick={ handlevent }></button>
        </>
    )
    */
    
}