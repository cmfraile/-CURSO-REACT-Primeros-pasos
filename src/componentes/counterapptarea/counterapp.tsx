import { useState } from 'react';
import PropTypes from 'prop-types';
import './counterapp.sass';

export const counter = ({contador}:any) => {

    const [ contadorhook , setCounter ] = useState(contador)
    const sc = (arg:string) => {
        switch(arg){
            case '+': setCounter(contadorhook+1);break;
            case '-': setCounter(contadorhook-1);break;
            case 'R': setCounter(contador);break;
        }
    };

    
    return(
        <div className='capp'>
            <h1>Counterapp</h1>
            <h2>{contadorhook}</h2>
            <button onClick={ () => {sc('+')} }>+1</button>
            <button onClick={ () => {sc('-')} }>-1</button>
            <button onClick={ () => {sc('R')} }>Reset</button>
        </div>
    )

    /*
    return(
        <>
            <h1>Counterapp</h1>
            <h2>{contador}</h2>
            <button onClick={ handlevent }></button>
        </>
    )
    */
    
}


counter.propTypes = {contador: PropTypes.number.isRequired};