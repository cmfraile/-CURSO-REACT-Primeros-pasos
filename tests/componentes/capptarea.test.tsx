import { render , screen } from '@testing-library/react';
import { counter as Capptarea } from '../../src/componentes/counterapptarea/counterapp';
import React from 'react';


const main = () => {

    test('Debe de hacer match con el snapshot',() => {
        expect(render(<Capptarea contador={100}/>).container).toMatchSnapshot();
    });

    /*
    test('debe de mostrar el valor inicial de 100 y mandarse por props',() => {
        render(<Capptarea contador={100}/>);
        //expect( getByText('100') )
    })
    */
}

describe('Pruebas en <capptarea>',main);