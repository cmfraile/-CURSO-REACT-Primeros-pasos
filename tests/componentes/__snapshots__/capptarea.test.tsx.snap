// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <capptarea> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="capp"
  >
    <h1>
      Counterapp
    </h1>
    <h2>
      100
    </h2>
    <button>
      +1
    </button>
    <button>
      -1
    </button>
    <button>
      Reset
    </button>
  </div>
</div>
`;
